### PR TITLE
Enable Dependabot updates for GitHub Actions

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR enables Dependabot, an automated agent provided for free from GitHub, to automatically check and update on a weekly basis GitHub Actions used by this project. GitHub Actions versions quickly become out of date, incurring security, performance, or other issues for projects. This is a preventative measure to help keep the project in good health. 

Note: these updates are performed in isolation from Dependabot security-focused updates, which are configured from GitHub settings for the repository. 

## What kind of change(s) are included?

- [ ] Documentation (changes docs or other related content)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] These changes pass all pre-commit checks.
- [x] I have added comments to my code to help provide understanding
- [ ] I have added a test which covers the code changes found within this PR
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# GitHub Dependabot configuration
+# Note: there is no interaction between this
+# configuration and dependabot security updates.
+# See here for more information:
+# https://docs.github.com/en/code-security/dependabot/dependabot-security-updates/about-dependabot-security-updates#about-dependabot-security-updates
+
+version: 2
```
Comment (jenna-tomkinson) [inline .github/dependabot.yml:7]: Is this the latest version, or is there a reason for a specific version? Just curious :)

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! I added one question but otherwise super nice small change~

File: .github/dependabot.yml (hunk 1 above)
Comment (d33bs) [reply .github/dependabot.yml:7]: Thanks, good question - this comes directly from examples and [GitHub's docs on the topic](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file):

`The file must start with version: 2`.

**Comment by d33bs** [comment]
Thanks @jenna-tomkinson ! Merging this in after responding to your comment.

**Pull request merged**